### PR TITLE
persistent node positions

### PR DESCRIPTION
**Pull request by naugtur**
Added persisting node positions to localStorage based on a hash of graph data source path and configuration. 
Nodes get saved a second after last mouse event in a row.
When saved, the data gets logged in the console too, so one can copy-paste it to the html file to store it cross-browser.

btw. removed underscore from html file as it's no longer needed.
I'd remove jQuery dependency too, but I'm lazy.

### CODE REVIEW

File: lib/dependo.js [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
     this.dependencies = madge(targetPath, this.config).tree;
 
-    if (this.config.transform && typeof (this.config.transform) == "function") {
+    if (this.config.transform && typeof (this.config.transform) == 'function') {
         this.dependencies = this.config.transform(this.dependencies);
     }
 
 }
 
-Dependo.prototype.generateHtml = function (options) {
```
Comment (naugtur) [inline lib/dependo.js]: `options` was never used

File: lib/html/d3-graph.js [hunk 2]
```diff
@@ -26,6 +26,7 @@
     $('path.link').removeAttr('data-show');
 
     nodes.style("stroke-opacity", function(o) {
+        var thisOpacity;
```
Comment (naugtur) [inline lib/html/d3-graph.js]: used to be global

File: lib/html/d3-graph.js [hunk 3]
```diff
@@ -167,22 +176,21 @@
...
 
       // Circles
       nodes.attr("class", function(d) { return formatClassName('node', d) })
       nodes.append("svg:circle")
         .attr("class", function(d) { return formatClassName('circle', d) })
         .attr("r", 6)
-        .on("mouseover", _.bind(onNodeMouseOver, this, nodes, links))
-        .on("mouseout", _.bind(onNodeMouseOut, this, nodes, links) );
+        .on("mouseover", onNodeMouseOver.bind(this, nodes, links) )
+        .on("mouseout", onNodeMouseOut.bind(this, nodes, links) );
```
Comment (naugtur) [inline lib/html/d3-graph.js]: say bye to IE8

File: package.json [hunk 4]
```diff
@@ -1,34 +1,39 @@
 {
-    "name": "dependo",
```
Comment (naugtur) [inline package.json]: after a `npm install` tis diff got big all of a sudden. I just added 2 keywords and one dependency

File: lib/html/d3-graph.js [hunk 5]
```diff
@@ -287,11 +296,43 @@
...
   }
+    
+  function persistNodes(nodes) {
+    var store = nodes.filter(function(n){ return n.fixed }).reduce(function(mem,n){
+      mem[n.id]=n;
+      return mem;
+    },{});
+    store = JSON.stringify(store);
+    window.localStorage.setItem("dependo"+window.graphIdentification, store);
+    console.log(store); //for manual saving
```
Comment (auchenberg) [inline lib/html/d3-graph.js]: Whoops :)

**Comment by auchenberg** [comment]
Looks good, remove the console.log, and I'll merge it in, thanks a bunch man!

File: lib/html/d3-graph.js (hunk 5 above)
Comment (naugtur) [inline lib/html/d3-graph.js]: That's actually a feature...
I can open console and copy the last entry, then put it in the html file in a field that's already prepared there.
We can iterate to make it nicer, but for now it's usable.

**Comment by naugtur** [comment]
Hi, you never responded to my justification for the console.log in there. 
If that's a strong *no* I can remove it and let's merge.

**Comment by auchenberg** [comment]
Sorry, I'm currently traveling Asia. I'd prefer if we keep console.logs out of the code, and find an better way to "dump" the data, if needed.

**Comment by naugtur** [comment]
Ok, there you go.
I'll see if I have time to make something nicer to dump the data.

**Comment by auchenberg** [comment]
LGTM :+1:

**Comment by auchenberg** [comment]
Thanks a bunch for the contribution!

**Pull request merged**